core/src/com/unciv/ui/popups/options/AdvancedTab.kt

```diff
@@ -98,29 +100,50 @@ class AdvancedTab(
             optionsPopup.reopenAfterDisplayLayoutChange()
         }
     }
-    
-    private fun addMaxAutosavesStored() {
+
+    private fun addAutosaveField() {
         add("Number of autosave files stored".toLabel()).left().fillX()
-        
-        val maxAutosavesStoredSelectBox = SelectBox<Int>(skin)
-        val maxAutosavesStoredArray = Array<Int>()
-        maxAutosavesStoredArray.addAll(1,2,5,10,15,20,35,50,100,150,200,250)
-        maxAutosavesStoredSelectBox.items = maxAutosavesStoredArray
-        maxAutosavesStoredSelectBox.selected = settings.maxAutosavesStored
-        
-        add(maxAutosavesStoredSelectBox).pad(10f).row()
-        
-        maxAutosavesStoredSelectBox.onChange {
-            settings.maxAutosavesStored = maxAutosavesStoredSelectBox.selected
-        }
+        val autosaveFieldTable = Table()
+        val autoSaveTrunsTextField = UncivTextField("",settings.maxAutosavesStored.toString())
+        autoSaveTrunsTextField.setTextFieldFilter { _, c -> c in " 1234567890" }
+        autosaveFieldTable.add(autoSaveTrunsTextField)
+        val autoSaveTrunsTextFieldButton = "Enter".toTextButton()
+
+        autosaveFieldTable.add(
+            autoSaveTrunsTextFieldButton.onClick {
+                if (!autoSaveTrunsTextField.text.isEmpty()) {
+                    val numberAutosaveTurns = autoSaveTrunsTextField.text.toInt()
+
+                    if (numberAutosaveTurns == 0) {
+                        val popup = Popup(stage)
+                        popup.addGoodSizedLabel("Autosave turns must be bigger than 0!", color = Color.RED)
+                        popup.addCloseButton()
+                        popup.open(true)
+
+                    } else if (numberAutosaveTurns >= 200) {
+                        val popup = Popup(stage)
+                        popup.addGoodSizedLabel(
+                            "Autosave turns that are bigger than 200 might take a lot of spcae\n" +
+                                "on your device.", color = Color.ORANGE)
+                        popup.addCloseButton()
+                        popup.open(true)
+                        settings.maxAutosavesStored = numberAutosaveTurns
+
+                    } else {
+                        settings.maxAutosavesStored = numberAutosaveTurns
+                    }
+                }
+            }
+        ).row()
+        add(autosaveFieldTable).row()
     }
-
+    
     private fun addAutosaveTurnsSelectBox() {
         add("Turns between autosaves".toLabel()).left().fillX()
 
         val autosaveTurnsSelectBox = SelectBox<Int>(skin)
         val autosaveTurnsArray = Array<Int>()
-        autosaveTurnsArray.addAll(1, 2, 5, 10)
+        autosaveTurnsArray.addAll(1, 2, 5, 10,15,20,50,100,1000)
```

Comment:
```suggestion
        autosaveTurnsArray.addAll(1, 2, 5, 10 ,20,50,100,1000)
```
Less than double is meaningless IMO